### PR TITLE
variable star patterns

### PR DESCRIPTION
**Pull request by GammaGames**
What does this implement/fix?
-----------------------------
I wanted to be able to set my own star patterns, like `··xxx+++++xx···············` and thought it would be helpful. This also lets users extend the `Stars` class and select a random pattern from a set without having to create a new `_Star` class.

Any other comments?
-------------------
It's pretty simple, you don't have to accept it I just thought it might be useful for people

### CODE REVIEW

**Comment by GammaGames** [comment]
It said 
```sh
File "/home/travis/build/peterbrittain/asciimatics/tests/test_effects.py", line 233, in <lambda>
    lambda value: self.assertIn(chr(value[0]), "·x+")))
AssertionError: ' ' not found in '·x+'
```
but the only values I have in my pattern are those three characters, there's no space in the pattern. Is this expected behavior?

File: asciimatics/effects.py [hunk 1]
```diff
@@ -464,19 +464,21 @@ class Stars(Effect):
     Add random stars to the screen and make them twinkle.
     """
 
-    def __init__(self, screen, count, **kwargs):
+    def __init__(self, screen, count, pattern="..+..   ...x...  ...*...         ", **kwargs):
         """
         :param screen: The Screen being used for the Scene.
         :param count: The number of starts to create.
+        :param pattern: The pattern for the stars to loop through
```
Comment (peterbrittain) [inline asciimatics/effects.py]: Can you please add that this must be a string?

File: tests/test_effects.py [hunk 2]
```diff
@@ -214,6 +214,31 @@ def test_stars(self):
...
+        self.assert_blank(canvas)
+        my_buffer = [[(32, 7, 0, 0) for _ in range(40)] for _ in range(10)]
+        for i in range(10):
+            effect.update(i)
+            self.assertTrue(self.check_canvas(
+                canvas,
+                my_buffer,
+                lambda value: self.assertIn(chr(value[0]), " TES")))
+
+        # Check there is no stop frame by default.
```
Comment (peterbrittain) [inline tests/test_effects.py]: There is no need to duplicate the rest of the tests in this function as they will not be affected by the custom pattern.  I'm trying to get more focused in what each test actually checks these days...  :-)

File: tests/test_effects.py [hunk 3]
```diff
@@ -214,6 +214,31 @@ def test_stars(self):
         event = object()
         self.assertEqual(event, effect.process_event(event))
 
+    def test_stars_pattern(self):
+        """
+        Check that Stars custom pattern value works works.
```
Comment (peterbrittain) [inline tests/test_effects.py]: Typo: please remove one "works".

**Comment by GammaGames** [comment]
What happened to the travis ci build? I can't decipher the output

**Comment by peterbrittain** [comment]
Good question...  I use travis and appveyor to get cross-platform coverage.  The last step of the travis process is to download the coverage artefacts from appveyor.  Occasionally this fails.  I've just kicked off those two jobs again.

**Comment by GammaGames** [comment]
[This one](https://travis-ci.org/peterbrittain/asciimatics/builds/502605477)? Looks like it passed this time, thank you!

**Comment by peterbrittain** [comment]
Yup.  You're welcome.

**Pull request merged**